server/src/main/java/org/elasticsearch/index/analysis/Analysis.java

```diff
@@ -78,19 +78,20 @@
 
 public class Analysis {
 
-    public static Version parseAnalysisVersion(Settings indexSettings, Settings settings, Logger logger) {
-        // check for explicit version on the specific analyzer component
-        String sVersion = settings.get("version");
-        if (sVersion != null) {
-            return Lucene.parseVersion(sVersion, Version.LATEST, logger);
-        }
-        // check for explicit version on the index itself as default for all analysis components
-        sVersion = indexSettings.get("index.analysis.version");
-        if (sVersion != null) {
-            return Lucene.parseVersion(sVersion, Version.LATEST, logger);
+    static void deprecateSetting(String settingName, Version version,
+                                 IndexSettings indexSettings, Settings settings, String componentName,
+                                 DeprecationLogger deprecationLogger) {
+        // check for the setting on the specific index component
+        // and on the index itself as default for all analysis components
+        if (settings.hasValue(settingName) || indexSettings.getSettings().hasValue("index.analysis." + settingName)) {
+            if (indexSettings.getIndexVersionCreated().before(version)) {
```

Comment:
Where does this logic comes from ? Is it something that you added for this pr ? We should only check that the version is parseable, the validation against the index created version looks like a new constraint that would break settings of users that set the wrong version (accepted today).

Reply:
hmm... from the code in your comment? :) I changed it to 
```
if (Version.CURRENT.before(version)) 
    // ...
```